mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -556,6 +579,28 @@ class SparseVector(
       i += 1
     }
   }
+
+  override def hashCode(): Int = {
```

Comment:
This is a contract in Java (http://en.wikipedia.org/wiki/Java_hashCode()):

> The general contract for overridden implementations of this method is that they behave in a way consistent with the same object's equals() method: that a given object must consistently report the same hash value (unless it is changed so that the new version is no longer considered "equal" to the old), and that two objects which equals() says are equal must report the same hash value.

So `sv == dv` -> `sv.## == dv.##`.